### DISCUSSION TITLE
HmIP-ESI-IEC und HmIP-ESI-GAS

### DISCUSSION THREAD

**Discussion by detLAN** [SukramJ/aiohomematic#1479] [Q&A]
Seit ca. einem Monat habe ich die Schnittstelle für Smart Meter mit Info-Schnittstelle (HmIP-ESI-IEC) im Einsatz:
![SCR-20240404-nqrf](https://github.com/danielperna84/hahomematic/assets/64675966/65111bbc-fb3f-4836-b080-71dd5036efb8)
<img width="1065" alt="SCR-20240404-nrgt" src="https://github.com/danielperna84/hahomematic/assets/64675966/95bfcbd9-22b2-4e3d-b2c5-31c8ad2b992f">

Heute habe ich die Schnittstelle für Gaszähler (HmIP-ESI-GAS) in Betrieb genommen.
<img width="785" alt="SCR-20240404-nqiu" src="https://github.com/danielperna84/hahomematic/assets/64675966/3d6f9a47-da24-4607-b6ac-d0a41304f683">
<img width="1065" alt="SCR-20240404-nrbc" src="https://github.com/danielperna84/hahomematic/assets/64675966/46f32bff-2826-45e8-a6fd-d2ad590eb13e">

Auf RaspMatic sieht soweit alles prima aus. Auf Home Assistant komme ich mit den Werten nicht zurecht:
<img width="323" alt="SCR-20240404-nxam" src="https://github.com/danielperna84/hahomematic/assets/64675966/7f92e0eb-8c93-43d0-b7bd-e725a2263c8f">
Hier habe ich: ch1 = Leistung, ch2 = Strombezug (gesamt), ch4 = Stromeinspeisung (gesamt)
<img width="324" alt="SCR-20240404-nxeh" src="https://github.com/danielperna84/hahomematic/assets/64675966/2a7f87f1-040f-4ad4-820a-9ef80c4f1077">
Hier habe ich: ch1 = Gas-Fluss, ch2 = Gas-Volumen (heute)

**Es fehlt: ch2 = Gasbezug (gesamt), welcher unknown ist und die falsche Einheit enthält. Wie oder wo kann ich dies korrigieren?**
<img width="560" alt="SCR-20240404-oauz" src="https://github.com/danielperna84/hahomematic/assets/64675966/9a15584d-3789-4e48-abb3-f58d0e7c52c0">

**Comment by SukramJ** [comment]
Da muss ich raten, weil ich selber keinen HmIP-ESI habe.

Der Wert ist unbekannt:
Ich würde mal den [Cache löschen](https://github.com/danielperna84/custom_homematic#homematicip_localclear_cache) und HA neustarten.
Danach ggf. das [Gerät aus HA](https://github.com/danielperna84/custom_homematic#homematicip_localdelete_device) löschen und HA neustarten.

**Reply by detLAN** [reply]
Vielen Dank für Deine Antwort. Wie geschrieben habe ich erst heute die Schnittstelle HmIP-ESI-GAS installiert.
Habe jetzt wie gewüsncht den Cache gelöscht und anschließend HA neu gestartet, jedoch ohne Erfolg.
Anschließend habe ich das Gerät gelöscht und anschließend HA neu gestartet, ebenfalls ohne Erfolg.

Ich weiss nicht wie HA bei der HmIP-ESI Integration zwischen IEC und GAS unterscheidet. 
Mit ch2 wird der Zählerstand **Bezogene Energie** übergeben, welcher bei GAS in m³ gemessen wird. Für diesen Wert nimmt HA jedoch kWh an, was evtl. zu ein **unknown** führt. Es gibt je nach Sensor unterschiedliche Einheiten für ch2. Dei IEC für Strombezug ist es kWh, bei GAS für Gasbezug müsste es m³ sein. Dies lässt sich jedoch nicht einstellen - siehe letzten Screenshot im ersten Post. Das **Gas-Volumen (heute)** entspricht dem rechnerischen Wert **Verbrauch heute** bei ch2 in der CCU (wobei HA immer 0,03 m³ zuviel anzeigt).

Es wird wie beim Strom auch beim Gas der Zählerstand ch2 benötigt, damit das Energie-Dashboard damit arbeiten kann. Der **heute** Wert wird tätglich auf Null zurück gesetzt.

Evtl. könnte die HA Integration eine Unterscheidung über ch1 herstellen?
<img width="927" alt="SCR-20240404-pguu" src="https://github.com/danielperna84/hahomematic/assets/64675966/0f84b113-5040-4ff4-808a-bcf910c4e99d">

**Reply by SukramJ** [reply]
Das ganze ist eine Reihe von Sensoren für die im Gegensatz zu `climate` oder `light` keine zusätzliche Programmierung benötigen.

> Ich weiss nicht wie HA bei der HmIP-ESI Integration zwischen IEC und GAS unterscheidet.

Bei Sensoren wird auch keine Unterscheidung gemacht. Es werden einfach die benötigten Messpunkte angezeigt.
Wenn die `device_class` oder `unit` nicht stimmt kann das durch den Benutzer über HA per UI oder [customize](https://www.home-assistant.io/docs/configuration/customizing-devices/) selber geändert werden.

Warum der Wert nicht angezeigt wird kann ich nicht sagen, aber einen Zusammenhang zur Einheit gibt es meines Wissens nach nicht.

> Evtl. könnte die HA Integration eine Unterscheidung über ch1 herstellen?

Ich werde auch zukünftig für Sensoren kein programatisches Customizing einbauen.

**Reply by detLAN** [reply]
> Bei Sensoren wird auch keine Unterscheidung gemacht. Es werden einfach die benötigten Messpunkte angezeigt.
Wenn die device_class oder unit nicht stimmt kann das durch den Benutzer über HA per UI oder [customize](https://www.home-assistant.io/docs/configuration/customizing-devices/) selber geändert werden.

Da komme ich nicht weiter, wo soll ich da die Einheit ändern können? Ich finde hier keine Einstellmöglichkeit.

Könnte es Sinn machen, wenn ich die Schnittstelle für Strom mal komplett lösche, also auch in der CCU und dann die Schnittstelle für Gas neu einlese? Klar, vorher dann natürlich auch in HA.

**Reply by SukramJ** [reply]
> Da komme ich nicht weiter, wo soll ich da die Einheit ändern können? Ich finde hier keine Einstellmöglichkeit.

Ich erwarte schon, das dir den Link anschaust. Im Abschnitt `Customizing entities` und [MANUAL CUSTOMIZATION](https://www.home-assistant.io/docs/configuration/customizing-devices/#manual-customization) ist alles beschrieben.

unit_of_measurement: m³
device_class: gas

**Comment by SukramJ** [comment]
@Boarder Wird bei deinem HmIP-ESI-GAS der Zählerstand/energy counter für Gas (Kanal 2) angezeigt?

**Comment by Boarder** [comment]
Folgende Sensoren habe ich:Am 04.04.2024 um 17:41 schrieb SukramJ ***@***.***>:﻿
@Boarder Wird bei deinem HmIP-ESI-GAS der Zählerstand/energy counter für Gas (Kanal 2) angezeigt?

**Reply by SukramJ** [reply]
Danke, aber da fehlt der Inhalt. (Bilder werden bei Mailantworten nicht angezeigt)

**Reply by Boarder** [reply]
Jetzt? ![image](https://github.com/danielperna84/hahomematic/assets/45979973/1026b125-598e-4645-8397-713a415fc2a6)

**Reply by detLAN** [reply]
@Boarder Danke für den Screenshot. Sieht ja genauso wie bei mir aus. Was macht Dein Gas Helper?
Bei der Schnittstelle für Strom wird der Zählerstand unter ch2 abgebildet. Unter ch2 sollte bei Gas eigentlich auch der Zählerstand abgebildet werden. Jedoch wird hier wie bei Dir unknown ausgegeben. Wie hast Du dies gelöst?

**Reply by SukramJ** [reply]
Am Besten mal mit dem [Homematic Manager](https://github.com/hobbyquaker/homematic-manager) oder wenn Ihr RaspberyyMatic benutzt über `Systemsteuerung/Dev Config/Device List` das Gerät öffnen und schauen, ob der gewünschte Wert auch über die Schnittstelle verfügbar ist. Ob der Wert über die UI verfügbar ist für mich nicht relevant.

**Comment by detLAN** [comment]
@SukramJ 
Danke für Deine Bemühungen. Hier die Ergebnisse als Screenshots. Etwas merkwürdig wie ich finde, eQ3 gibt beim Strom die **Zählerstande** aus und bei Gas _nur_ den **Verbrauch heute**. Somit können wir in HA den Zählerstand für Gas nicht auslesen.
![strom](https://github.com/danielperna84/hahomematic/assets/64675966/3d66f28f-ca43-44a6-9be3-3d879ee8d83e)
![gas](https://github.com/danielperna84/hahomematic/assets/64675966/5e2cfafd-c71e-432a-8812-c9c4d0128a44)

@Boarder Mich würde interessieren wofür Du einen Gas Helper eingerichtet hast? Errechnest Du Dir damit den Zählerstand Deines Gaszählers? Könntest Du den Helper hier offen legen? DANKE

**Comment by Boarder** [comment]
Den Gas Helper hatte ich angelegt als das Gerät noch nicht in das Energie Dashboard eingebunden werden konnte, hab ihn jetzt auch wieder gelöscht. Am 04.04.2024 um 22:27 schrieb detLAN ***@***.***>:﻿
@SukramJ
Danke für Deine Bemühungen. Hier die Ergebnisse als Screenshots. Etwas merkwürdig wie ich finde, eQ3 gibt beim Strom die Zählerstande aus und bei Gas nur den Verbrauch heute. Somit können wir in HA den Zählerstand für Gas nicht auslesen.
strom.jpg (view on web)
gas.jpg (view on web)
@Boarder Mich würde interessieren wofür Du einen Gas Helper eingerichtet hast? Errechnest Du Dir damit den Zählerstand Deines Gaszählers? Könntest Du den Helper hier offen legen? DANKE

**Comment by Baxxy13** [comment]
Für den Gaszähler gibt es nur 2 relevante Datenpunkte (Parameter) im Gerät:
Kanal:1 - GAS_FLOW
Kanal:2 - GAS_VOLUME

Man muss hier beachten das IEC und GAS zwei komplett verschiedene Energiezähler darstellen.
- IEC bekommt über das Protokoll alle Daten inkl. echtem Zählerstand übermittelt. Das was die Messtelle ausgibt wird entsprechend in der WebUI angezeigt.
- GAS bekommt nur Impulse und errechnet mittels der Gerätekonstante den Durchfluss (GAS_FLOW), außerdem erhöht das Gerät mittels der Impulse seinen eigenen (flüchtigen) Zählerstand.

Beispiel:
Bei Inbetriebnahme ist der interne Zählerstand des ESI (GAS) "0". Bei eingestellten 0.01m³/imp erreicht der Zählerstand nach 100 Impulsen 1.0m³.

Spekulation:
Die WebUI-Anzeige "Zählerstand" dürfte eine Systemvariable sein die per systeminternem Programm berechnet wird. Hier wird vereinfacht gesagt der interne Zählerstand zum eingestellten Startwert hinzugerechnet.

Bei den älteren Energiezählern gab es immer "Energiezähler-Gerät" und "Energiezähler-CCU/RM". 
Möglicherweise hat sich eQ-3 dazu entschieden "Energiezähler-Gerät" wegzulassen da dieser Zähler keine Relevanz hat.
Einmal Strom weg/Batterie wechseln und der begann bei "0".

Grundsätzlich sollte GAS_VOLUME stetig steigen (Gasverbrauch vorausgesetzt) und nur bei Batteriewechsel auf "0" zurückfallen. 
Damit müsste er sich doch im Energy-Dashboard nutzen lassen. Relevant ist doch nur der Verbrauch, nicht irgendein Zählerstand.

**Reply by detLAN** [reply]
@Baxxy13 Vielen Dank für Deine ausfürlich Erklärung. Die IEC Schnittstelle für Strom habe ich schon ein paar Wochen hier im Einsatz. Diese liefert die aktuellen Zählerstände und HA kann im Energie-Dashboard damit umgehen. Jetzt habe ich diese Woche die ESI für Gas bekommen und war verwundert, dass ich aufeinmal den aktuellen Tageswert angeben muss. Und da der Wert in HA immer um 0,03 m³ höher ausfällt als in der CCU, dachte ich, dass ich den Gas Zählerstand angeben muss, welchen ich in HA nicht habe.

Zu Deiner Spekulation: 
Für einen ESI werden vier Systeminterne Programme angelegt, zwei für CH2 und je eines für CH3 und CH4. Analog dazu gibt es dann vier Systeminterne Variablen.

Edit:
Und Dank dieses Hinweis (Spekulation) von Dir, hab ich nun natürlich auch den Zählerstand des Gaszähler als Entität - Danke! Beim Energie-Dashboard belasse ich es jetzt aber ersteinmal beim aktuellen Tageswert für Gas.

**Reply by deGrasi** [reply]
> Und Dank dieses Hinweis (Spekulation) von Dir, hab ich nun natürlich auch den Zählerstand des Gaszähler als Entität - Danke! 
@detLAN kannst du mir sagen, wie du das gemacht hast?

**Reply by detLAN** [reply]
@deGrasi Das ist ganz einfach, suche bei den Entitäten nach der Seriennummer - Filter Deaktiviert verwenden. Und dann musst Du eine der beiden mit Endung :2 aktivieren - bei mir war es die untere.

![pic](https://github.com/danielperna84/hahomematic/assets/64675966/ad5ea4d3-07df-4643-a6cd-bb81b2c3bea5)

**Reply by deGrasi** [reply]
@detLAN 
Vielen lieben Dank für deine Rückmeldung. Das hat nun funktioniert und die Entität ist sichtbar. 
Allerdings zur Einbindung in ein Dashboard ist allerdings wieder ein Helfer notwendig, da die Entität selbst nicht ausgewählt werden kann. Sehe ich das richtig?

**Reply by detLAN** [reply]
@deGrasi Genau so ist es. Allerdings brauchst Du für das Energie-Dashboard den Wert nicht, es reicht der Tageswert. Ich lasse mir die Zählerstände auf einer Übersichtsseite anzeigen, damit ich nicht in den Keller muss, wenn ich die Zählerstände Melden muss.

**Reply by deGrasi** [reply]
@detLAN Genau das ist auch mein Gedanke, die Zählerstände übersichtlich an einem Ort zu haben.
Vielen Dank für deine Unterstützung.

**Reply by Baxxy13** [reply]
Es müsste doch auch gehen wenn man die SysVar-Entität customized so daß sie dann konform für das Energie-Dashboard ist.

**Comment by Isengo1989** [comment]
Mal eine andere Frage zum Helfer. Ist das irgendwie über die GUI möglich oder muss ich was custom bauen?
![2024-10-25_09-16_1](https://github.com/user-attachments/assets/7888c725-dfb9-42e6-8f5a-9c010aab9b8c)
![2024-10-25_09-16](https://github.com/user-attachments/assets/1fb008bf-8324-4426-b00d-62d095a20455)
![2024-10-25_09-14](https://github.com/user-attachments/assets/2c2991d6-cc86-476f-8db7-8784c49a6c74)


Würde gerne den Zählerstand auch von HA auslesen (obwohl die Genauigkeit leider zu wünschen übrig lässt, ist das bei euch auch so?)
![2024-10-25_09-13](https://github.com/user-attachments/assets/dfe454a7-6b34-42be-9a66-098cb549d104)
